sway/desktop/render.c

```diff
@@ -267,6 +269,10 @@ static void render_saved_view(struct sway_view *view,
 
 	render_texture(wlr_output, damage, view->saved_buffer->texture,
 			&box, matrix, alpha);
+
+	if (view->surface) {
+		wlr_presentation_surface_sampled(server.presentation, view->surface);
```

Comment:
Hmm. I don't think this is correct, because it displays a previous version of the view. So sending presented for this surface makes the client think its last update has been presented,

Reply:
Where's the correct place to mark the surface as sampled in this case? Probably somewhere where the buffer is saved?

Reply:
Actually I guess in this case the saved buffer needs to store the presentation feedback object associated with the commit from which the buffer was saved, and then either discard it if it's overridden with another saved buffer, or signal it as sampled in this spot.

Reply:
I tried approaching this problem by adding a third "saved" state to the presentation feedback. The idea was that a feedback can become "saved" and then saved feedbacks could be either sampled or discarded (in addition to directly sampling feedbacks). I would then mark the feedback as "saved" in `view_save_buffer`, sample the "saved" feedback in `render_saved_view` and discard the "saved" feedback in `view_remove_saved_buffer`. However, I could not get it to work, because:
- when the surface is dragged around as a floating window, `view_save_buffer` and `view_remove_saved_buffer` are called multiple times on the same buffer, which leads to the feedback being discarded in my model;
- when the surface is first created and committed I observed that it's possible for `view_save_buffer` to be called before the commit handler of the feedback, which means that the feedback that should have been "saved" by `view_save_buffer` wasn't "saved" because it wasn't "committed" yet at that point in time.

I'm not sure how else to approach this problem in a way which is reasonable. Any help would be appreciated.